src/dag-node/dagNode.js

```diff
@@ -27,16 +27,10 @@ class DAGNode {
     })
     links = sortLinks(links)
 
-    if (serializedSize === 0) {
-      serializedSize = serializeDAGNode({
-        Data: data,
-        Links: links
-      }).length
-    }
-
     this._data = data
     this._links = links
-    this._serializedSize = serializedSize
+    this._serializedSize = typeof serializedSize === 'number' ? serializedSize : null
```

Comment:
`serializedSize` isn't manipulated within the constructor. Would it work to change its signature to `constructor (data, links = [], serializedSize = null)` and then do a `this._serializedSize = serializedSize` here?

Reply:
I did this because you have no guarantees of type even with a default argument value, in this way, it's either `null` or a number, no options for anything else. It's not a big deal, I just prefer more assurance of type and default values don't give you much of that. Your call.

Reply:
I see. It find it strange if you have a default value as the second parameter, but no default value for the third. So I would prefer having at least a `serialized = null` in the constructor.

When reading the code I didn't get that it was there as a sanity check. We have some checks at the top, so I would rather put a check like

```js
if (serializedSize !== null && serializedSize !== number) {
   throw new Error('Passed \'serializedSize\' must be a number')
}
```

No sure about this though, so take this as input and I you "making the call" back to you.

Reply:
I like JS APIs to be very explicit about types so I'm +1 on this, it's just a bit more "breaking" than allowing a silent ignore. But I suppose that third argument isn't intended for public use so it's safe to be brutal about it?